tests/integration/buildcmd/test_build_cmd.py

```diff
@@ -402,3 +402,83 @@ def _verify_built_artifact(self, build_dir, function_logical_id, expected_files)
         all_artifacts = set(os.listdir(str(resource_artifact_dir)))
         actual_files = all_artifacts.intersection(expected_files)
         self.assertEquals(actual_files, expected_files)
+
+
+class TestBuildCommand_SingleFunctionBuilds(BuildIntegBase):
+    template = "many-functions-template.yaml"
+
+    EXPECTED_FILES_GLOBAL_MANIFEST = set()
+    EXPECTED_FILES_PROJECT_MANIFEST = {'__init__.py', 'main.py', 'numpy',
+                                       # 'cryptography',
+                                       "jinja2",
+                                       'requirements.txt'}
+
+    def test_fucntion_not_found(self):
+        overrides = {"Runtime": 'python3.7', "CodeUri": "Python", "Handler": "main.handler"}
+        cmdlist = self.get_command_list(parameter_overrides=overrides,
+                                        function_identifier="FunctionNotInTemplate")
+
+        process = subprocess.Popen(cmdlist, cwd=self.working_dir, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+
+        self.assertEquals(process.returncode, 1)
+        self.assertIn('FunctionNotInTemplate not found', str(stderr.decode('utf8')))
+
+    @parameterized.expand([
+        ("python3.7", False, "FunctionOne"),
+        ("python3.7", "use_container", "FunctionOne"),
+        ("python3.7", False, "FunctionTwo"),
+        ("python3.7", "use_container", "FunctionTwo")
+    ])
+    def test_build_single_function(self, runtime, use_container, function_identifier):
+        # Don't run test on wrong Python versions
+        py_version = self._get_python_version()
+        if py_version != runtime:
+            self.skipTest("Current Python version '{}' does not match Lambda runtime version '{}'".format(py_version,
+                                                                                                          runtime))
+
+        overrides = {"Runtime": runtime, "CodeUri": "Python", "Handler": "main.handler"}
```

Comment:
This test is meant to test that we can build a single function in a template of multiple. I really only cared about the single function being built and not that is built the exact one. I am assuming that our other tests are covering this area. So instead of creating more scaffolding, I bootstrapped the these new tests to use the same scaffolding we already have.

Reply:
Yeah i can understand that, what i was trying to get behind was that a customer will not create two resources named differently that reference the same handler. but given that the built artifacts are going to separate build folders, its okay in this case.